openshift-kube-apiserver/filters/deprecatedapirequest/apiaccess_count_controller.go

```diff
@@ -94,24 +95,40 @@ func (c *controller) Start(stop <-chan struct{}) {
 	}()
 
 	// write out logs every c.updatePeriod
-	go wait.NonSlidingUntilWithContext(ctx, c.persistRequestCountForAllResources, c.updatePeriod)
+	go wait.NonSlidingUntilWithContext(ctx, c.sync, c.updatePeriod)
+}
+func (c *controller) sync(ctx context.Context) {
+	currentHour := time.Now().Hour()
+	c.persistRequestCountForAllResources(ctx, currentHour)
 }
 
-func (c *controller) persistRequestCountForAllResources(ctx context.Context) {
-	klog.V(2).Infof("updating top APIRequest counts")
-	defer klog.V(2).Infof("finished updating top APIRequest counts")
+func (c *controller) persistRequestCountForAllResources(ctx context.Context, currentHour int) {
+	klog.V(4).Infof("updating top APIRequest counts")
+	defer klog.V(4).Infof("finished updating top APIRequest counts")
 
 	// get the current count to persist, start a new in-memory count
 	countsToPersist := c.resetRequestCount()
 
 	// remove stale data
-	expiredHour := (time.Now().Hour() + 1) % 24
-	currentHour := time.Now().Hour()
+	expiredHour := (currentHour + 1) % 24
 	countsToPersist.ExpireOldestCounts(expiredHour)
 
 	// when this function returns, add any remaining counts back to the total to be retried for update
 	defer c.requestCounts.Add(countsToPersist)
 
+	// Add resources that have an existing APIRequestCount so that the current and hourly logs
+	// continue to rotate even if the resource has not had a request since the last restart.
```

Comment:
explain why this works for the next hour too.

Reply:
Updated doc.